cogeo_mosaic/backends/file.py

```diff
@@ -0,0 +1,66 @@
+"""cogeo-mosaic File backend."""
+
+from typing import Dict, List, Optional, Union
+
+import json
+import functools
+
+import mercantile
+
+from cogeo_mosaic.model import MosaicJSON
+from cogeo_mosaic.backends.base import BaseBackend
+from cogeo_mosaic.backends.utils import (
+    _compress_gz_json,
+    _decompress_gz,
+    get_assets_from_json,
+)
+
+
+class FileBackend(BaseBackend):
+    """Local File Backend Adapter"""
+
+    def __init__(self, path: str, mosaic_def: Optional[Union[MosaicJSON, Dict]] = None):
+        """Initialize FileBackend."""
+        self.path = path
+
+        if mosaic_def is not None:
+            self.mosaic_def = MosaicJSON(**dict(mosaic_def))
+        else:
+            self.mosaic_def = self.read()
+
+    def tile(self, x: int, y: int, z: int) -> List[str]:
+        """Retrieve assets for tile."""
+        return get_assets_from_json(self.mosaic_def.tiles, self.quadkey_zoom, x, y, z)
+
+    def point(self, lng: float, lat: float) -> List[str]:
+        """Retrieve assets for point."""
+        tile = mercantile.tile(lng, lat, self.quadkey_zoom)
+        return get_assets_from_json(
+            self.mosaic_def.tiles, self.quadkey_zoom, tile.x, tile.y, tile.z
+        )
+
+    def write(self, gzip=None):
+        """Write mosaicjson document to a file."""
+        body = dict(self.mosaic_def)
+        if gzip or (gzip is None and self.path.endswith(".gz")):
```

Comment:
@kylebarron @geospatial-jeff I think I mis understood this one.

if an user wants to create a mosaic file, basically I was expecting we could do:

```python
with MosaicBackend(None, mosaic_def=mosaic_content) as mosaic:
    mosaic.write("mosaic.json")
```
This won't work because self.path is set to `None` and the function doesn't allow `"mosaic.json"` as input. 

for now the correct way to do is: 

```python
with MosaicBackend("mosaic.json", mosaic_def=mosaic_content) as mosaic:
    mosaic.write()
```

any though on what's the best way to go forward ?

Reply:
> ```python
> with MosaicBackend("mosaic.json", mosaic_def=mosaic_content) as mosaic:
>     mosaic.write()
> ```

I think this is fine, because it mimics `open()`

Reply:
ok so we will need to add a test to make sure `path` is set to something

Reply:
> ok so we will need to add a test to make sure path is set to something

well this might not be needed, what if someone just want to create a mosaic in memory without reading it (ok this is an edge case), but we might request a path will it won't be used 
🤔

Reply:
> what if someone just want to create a mosaic in memory without reading it

I think that's a perfect example for a custom backend. If someone wants that, they can have a "backend" that writes to a `BytesIO` or `StringIO` buffer.